src/Symfony/Component/Routing/Route.php

```diff
@@ -123,6 +123,19 @@ public function getPath()
      */
     public function setPath($pattern)
     {
+        if (false !== strpbrk($pattern, '?<')) {
+            $pattern = preg_replace_callback('#\{(\w++)(<.*?>)?(\?[^\}]*+)?\}#', function ($m) {
```

Comment:
what does `{bar<>}` exactly do?

Reply:
this is already tested elsewhere, nothing specific to this feature